flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamTaskNonSelectableInputThroughputBenchmark.java

```diff
@@ -0,0 +1,122 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.io.benchmark;
+
+import org.apache.flink.runtime.io.network.partition.consumer.IterableInputChannel;
+import org.apache.flink.streaming.runtime.io.StreamTwoInputProcessor;
+import org.apache.flink.streaming.runtime.io.benchmark.StreamTaskInputBenchmarkEnvironment.ProcessorAndChannels;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Task-input (non-selectable) throughput benchmarks executed by the external
+ * <a href="https://github.com/dataArtisans/flink-benchmarks">flink-benchmarks</a> project.
+ */
+public class StreamTaskNonSelectableInputThroughputBenchmark extends StreamTaskInputThroughputBenchmarkBase {
+
+	public void setUp(int numInputGates, int numChannelsPerGate, long numRecordsPerChannel) throws Exception {
+		setUp(
+			numInputGates, numInputGates,
+			numChannelsPerGate, numChannelsPerGate,
+			numRecordsPerChannel, numRecordsPerChannel,
+			new SummingLongStreamOperator());
+	}
+
+	@Override
+	protected AbstractTaskInputProcessorThread createProcessorThread(
+		int numInputGates1,
+		int numInputGates2,
+		int numChannels1PerGate,
+		int numChannels2PerGate,
+		long numRecords1PerChannel,
+		long numRecords2PerChannel,
+		long inputValue1,
+		long inputValue2,
+		SummingLongStreamOperator streamOperator) throws IOException {
+
+		ProcessorAndChannels<StreamTwoInputProcessor<?, ?>, IterableInputChannel> processorAndChannels =
+			environment.createTwoInputProcessor(
+				numInputGates1,
+				numInputGates2,
+				numChannels1PerGate,
+				numChannels2PerGate,
+				numRecords1PerChannel,
+				numRecords2PerChannel,
+				1,
+				2,
+				streamOperator);
+
+		return new StreamTwoInputProcessorThread(
+			processorAndChannels.processor(),
+			processorAndChannels.channels(),
+			streamOperator);
+	}
+
+	// ------------------------------------------------------------------------
+	//  Utilities
+	// ------------------------------------------------------------------------
+
+	private static class StreamTwoInputProcessorThread extends AbstractTaskInputProcessorThread {
+
+		private final StreamTwoInputProcessor<?, ?> inputProcessor;
+
+		private final SummingLongStreamOperator streamOperator;
+
+		private volatile boolean continuousProcessing = true;
```

Comment:
`continuousProcessing` is actually equivalent to `running` in `TwoInputStreamTask#run()` and it is just right used to stop benchmarks.  About using `EndOfPartitionEvent`, reconstructing `StreamTwoInputProcessor`  and `inputChannels`  may make some JIT optimization to be ineffective and fail to achieve iterative warmup effect.

Reply:
I'm not sure if this would affect JIT optimisations, since the optimisations are not tied up to the specific class instances. The issue might be that constructing `StreamTwoInputProcessor` and `inputChannels` might take some time, but those are quick operations, minuscule compared to a 1 second benchmark run.

There is a simple way to verify those things - just keep increasing the number of processed records per iteration until adding more records do not improve the throughput.

Reply:
Personally, I think the current implementation is better because it avoids the impact of benchmark's own code as much as possible. Although the overhead of constructing `StreamTwoInputProcessor` and `inputChannels` with using `EndOfPartitionEvent` can be balanced to some extent by increasing the number of processed records, but it also will increase the time of benchmark. Using `EndOfPartitionEvent` doesn't seem to have any other benefits except that the code looks more beautiful. 

(The previous version is the use of `EndOfPartitionEvent` and I change it to the currently implementation.)

Reply:
still there is a performance downside, that you are not benchmarking the `StreamTwoInputProcessor` but also this `volatile`. If it's supposed to be an equivalent of `running` from task, we should benchmark the task class itself, not imitate it by copy/pasting the code. If the implementation of `Task` changes and we drop `volatile boolean running` there, this `volatile boolean continuousProcessing` would be outdated. In other words, we should either:

1. benchmark Task classes, and that would include any potential existing `volatile boolean running`
2. benchmark processors, but in that case we shouldn't be adding extra overhead.

I guess that if we drop the outer loop that we were discussing in other comment, there won't be a need for this `volatile continuousProcessing` flag and we can safely benchmark processors without adding extra overhead, right?

Reply:
> I guess that if we drop the outer loop that we were discussing in other comment, there won't be a need for this volatile continuousProcessing flag and we can safely benchmark processors without adding extra overhead, right?

Yes, it's right.